src/main/java/ch/bfh/bti7081/s2015/red/PatientApp/View/MedicationIndexView.java

```diff
@@ -4,47 +4,68 @@
 
 import com.vaadin.annotations.Theme;
 import com.vaadin.annotations.Widgetset;
+import com.vaadin.navigator.Navigator;
 import com.vaadin.navigator.ViewChangeListener.ViewChangeEvent;
+import com.vaadin.ui.Button;
 import com.vaadin.ui.Label;
 import com.vaadin.ui.Button.ClickEvent;
+import com.vaadin.ui.Notification;
 
 import ch.bfh.bti7081.s2015.red.PatientApp.Model.MedicationEntry;
+import ch.bfh.bti7081.s2015.red.PatientApp.Presenter.MedicationIndexPresenter;
+import ch.bfh.bti7081.s2015.red.PatientApp.Presenter.MedicationPresenter;
 
 
 @Theme("mytheme")
 @Widgetset("ch.bfh.bti7081.s2015.red.PatientApp.MyAppWidgetset")
 public class MedicationIndexView extends BaseView<MedicationEntry>
 {
-	
-
+	private MedicationIndexPresenter entries = new MedicationIndexPresenter(null);
+	private NavigatorUI navigation = new NavigatorUI();
 
 	@Override
 	public void enter(ViewChangeEvent event) {
+
+		for (MedicationEntry medication : entries.getCollection()) {
```

Comment:
Move this to the `update` or `constructor` method.
`enter` is called on view change. If you use back/forward in your browser this will happen:
![screen shot 2015-05-12 at 18 41 44](https://cloud.githubusercontent.com/assets/1375745/7592962/e4b009e2-f8d6-11e4-84ab-37770521661e.png)